pkg/cmd/server/origin/master.go

```diff
@@ -273,6 +271,54 @@ func initAPIVersionRoute(root *restful.WebService, version string) {
 		Consumes(restful.MIME_JSON))
 }
 
+// If we know the location of the asset server, redirect to it when / is requested
+// and the Accept header supports text/html
+func assetServerRedirect(handler http.Handler, assetPublicURL string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		accept := req.Header.Get("Accept")
+		if req.URL.Path == "/" && strings.Contains(accept, "text/html") {
+			http.Redirect(w, req, assetPublicURL, http.StatusFound)
+		} else {
+			// Dispatch to the next handler
+			handler.ServeHTTP(w, req)
+		}
+	})
+}
+
+// TODO We would like to use the IndexHandler from k8s but we do not yet have a
+// MuxHelper to track all registered paths
+func indexAPIPaths(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path == "/" {
+			object := api.RootPaths{Paths: []string{
```

Comment:
Also, add a TODO in the integration test file to add a test when the asset config is nil, once David's changes to let you manipulate the config are available